Apps/Database/src/Delegates/DBProfileDelegate.cs

```diff
@@ -160,16 +160,20 @@ public int GetRegisteredUsersCount()
         }
 
         /// <inheritdoc />
-        public int GetLoggedInUsersCount(TimeSpan offset)
+        public IDictionary<DateTime, int> GetDailyLoggedInUsersCount(TimeSpan offset)
         {
-            DateTime now = DateTime.UtcNow;
-            DateTime clientTime = DateTime.SpecifyKind(now.Add(offset), DateTimeKind.Unspecified);
-            DateTimeOffset clientStartDate = new DateTimeOffset(clientTime.Date, offset);
-            DateTime queryStartTime = clientStartDate.UtcDateTime;
-            DateTime queryEndTime = now;
-            int result = this.dbContext.UserProfile
-                .Count(u => u.LastLoginDateTime >= queryStartTime && u.LastLoginDateTime < queryEndTime);
-            return result;
+            var unionResult = this.dbContext.UserProfile
+                .Select(x => new { HdId = x.HdId, lastLoginDate = x.LastLoginDateTime })
+                .Union(this.dbContext.UserProfileHistory
+                .Select(x => new { HdId = x.HdId, lastLoginDate = x.LastLoginDateTime })).ToList();
+
+            Dictionary<DateTime, int> dateCount = unionResult.Select(x => new
+            {
+                HdId = x.HdId,
+                lastLoginDate = x.lastLoginDate.Add(offset).Date,
+            }).Distinct().GroupBy(x => x.lastLoginDate).OrderBy(x => x.Key).ToDictionary(x => x.Key, x => x.Count());
```

Comment:
nice query!

Reply:
This doesn't look right, this looks like it will query and bring the entire user profile and user profile history tables back into memory and then does c# side query to filter the results.  This will be a no go for the Production DB, can you confirm?

Reply:
at this point, only the hdid and lastLoginDate are loaded. Since the offset needs to be executed on the c# side the queries are separated. I tried executing the date offset calculation on the first query, but EF could not translate it

Reply:
Updated query!